### PR TITLE
Don't validate a block announcement when syncing

### LINKED ISSUE #150: Don't validate a block announcement when syncing
When we sync the Parachain/Relay chain, we should not validate block announcements, because it leads to the following output:

```
2020-07-16 21:21:48 💔 Block announcement validation errored: UnknownBlock: 0x06b0…78b4
2020-07-16 21:21:48 💔 Block announcement validation errored: UnknownBlock: 0x06b0…78b4
```

This happens because the relay chain block is not yet known to the node.

### PR DESCRIPTION
**Pull request by cecton**
Fixes #150

### CODE REVIEW

File: network/src/lib.rs [hunk 1]
```diff
@@ -114,6 +124,10 @@ where
 			});
 		}
 
+		if (self.is_major_syncing)() {
+			return Ok(Validation::Success { is_new_best: true });
+		}
```
Comment (bkchr) [inline network/src/lib.rs:129]:
```suggestion
		if (self.is_major_syncing)() {
			return Ok(Validation::Success { is_new_best: false });
		}
```

And please move it above the other if and remove the new added inner check.

Comment (cecton) [reply network/src/lib.rs]: done

File: network/src/lib.rs [hunk 2]
```diff
@@ -55,14 +55,20 @@ pub struct JustifiedBlockAnnounceValidator<B, P> {
 	phantom: PhantomData<B>,
 	polkadot_client: Arc<P>,
 	para_id: ParaId,
+	is_major_syncing: Box<dyn FnMut() -> bool + Send + Sync>,
```
Comment (bkchr) [inline network/src/lib.rs:58]: Why is that only a FnMut? Why not use the `SyncOracle`?

And please name it:

`polkadot_sync_oracle`.

Comment (cecton) [reply network/src/lib.rs:58]: :thinking: fair point, you're totally right

Comment (cecton) [reply network/src/lib.rs]: Can you check again? 6dec62e

Comment (cecton) [reply network/src/lib.rs]: I think it is possible to not use Box if I expect the polkadot's "Service" (the network) instead of a SyncOracle

Comment (bkchr) [reply network/src/lib.rs]: LGTM

**Pull request merged**